--- a/Makefile	Sun Jul 05 03:07:25 2015
+++ b/Makefile	Sun Jul 05 03:27:02 2015
@@ -104,7 +104,9 @@
 	mkdir -p $(DESTDIR)$(INSTALLBINDIR)
 	mkdir -p $(DESTDIR)$(INSTALLLIBDIR)
 	mkdir -p $(DESTDIR)$(INSTALLINCDIR)
+	mkdir -p $(DESTDIR)$(INSTALLLIBDIR)/pkgconfig/
 	$(INSTALL) -m 644 include/*.h $(DESTDIR)$(INSTALLINCDIR)
+	$(INSTALL) -m 755 flite.pc $(DESTDIR)$(INSTALLLIBDIR)/pkgconfig/
 	@ $(MAKE) -C main --no-print-directory DESTDIR=$(DESTDIR) install
 
 time-stamp :
