--- abxtest-dist	2004-03-19 15:20:51.677624467 +0100
+++ abxtest	2004-03-19 15:44:01.719279507 +0100
@@ -94,7 +94,7 @@
 
 echo "randomizing ..."
 
-tmp="/tmp/abx.$$"
+tmp=$(/bin/mktemp /tmp/abx.XXXXXX) || { echo "$0: can not create temporary file" >& 2; exit 1; }
 trap "rm -f $tmp" 0
 
 rand="${RANDOM_FILE:-/dev/random}"
@@ -131,7 +131,7 @@
 done
 
 exec 3<&-
-rm -f $tmp
+echo -n > $tmp
 
 probability() {
     bc <<EOF
@@ -226,7 +226,7 @@
 	    status=$?
 	    if [ $status -eq 0 -o $status -ge 128 ]
 	    then
-		rm -f $tmp
+	        echo -n > $tmp
 		trap - 2
 	    else
 		cat >&2 $tmp
@@ -261,6 +261,8 @@
     esac
 done
 
+rm -f $tmp
+
 echo
 echo "$trials trials completed"
 echo "A = $A"
