From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Thu, 3 Jan 2013 18:31:33 +0100
Subject: [PATCH] Fix -Wstrict-aliasing compiler warnings


diff --git a/cli/md5.c b/cli/md5.c
index 1700240..a9954c2 100644
--- a/cli/md5.c
+++ b/cli/md5.c
@@ -151,8 +151,7 @@ void MD5Final(digest, ctx)
     byteReverse(ctx->in, 14);
 
     /* Append length in bits and transform */
-    ((uint32 *) ctx->in)[14] = ctx->bits[0];
-    ((uint32 *) ctx->in)[15] = ctx->bits[1];
+    memcpy(ctx->in + 14 * sizeof (uint32), ctx->bits, 2 * sizeof (uint32));
 
     MD5Transform(ctx->buf, (uint32 *) ctx->in);
     byteReverse((unsigned char *) ctx->buf, 4);
